packages/preview-middleware-client/src/cpe/outline/nodes.ts

```diff
@@ -60,25 +105,77 @@ export async function transformNodes(input: OutlineViewNode[], scenario: Scenari
                 name: text ?? technicalName,
                 editable,
                 visible: current.visible ?? true,
-                children: await transformNodes(children, scenario)
+                children: await transformNodes(children, scenario, uniqueIDs)
             };
 
             items.push(node);
         }
 
         if (scenario === 'ADAPTATION_PROJECT' && current?.type === 'extensionPoint') {
+            const extensionPointInfo = (current as unknown as ExtendedOutlineNode).extensionPointInfo;
+
+            const hasDefaultContent = extensionPointInfo?.defaultContent.length > 0;
+            const hasCreatedControls = extensionPointInfo?.createdControls.length > 0;
+
+            let children: OutlineNode[] = [];
+
+            if (hasDefaultContent) {
+                extensionPointInfo?.defaultContent.forEach((id) => {
+                    uniqueIDs.add(id);
+                    addChildToExtensionPoint(id, children);
```

Comment:
<img width="343" alt="Screenshot 2024-05-10 at 12 52 59" src="https://github.com/SAP/open-ux-tools/assets/39880364/fc0eaa94-4c14-4308-be12-f0e1a3001b37">

I'll explain it by referencing this screenshot. Extension points are usually under some control, in this case, it is Panel. We can see that Panel has two extension points. ExtensionPoint2 has no default content nor created controls ExtensionPoint3 has default content and we use `addChildToExtensionPoint` to add children to it. defaultContent is an array of control IDs, therefore we need to construct those children ourselves. 

When you add a fragment at ExtensionPoint3 (a button for example) it will no longer display the default content and it will be an empty array. On the other hand, created controls will have an array with IDs, in our case, the id of the button.
<img width="327" alt="Screenshot 2024-05-10 at 13 28 37" src="https://github.com/SAP/open-ux-tools/assets/39880364/93edac91-7d77-444b-9812-ad455d6d4003">

Both default content and created controls are controls and have a parent so if we do not remove them they will be rendered on the same level as the two extension points and one Label under Panel. We want to visualize them under the extension point to which they belong. That is why we need to remove those duplicate nodes and that is why we keep a unique set of IDs to later remove them.